servers/cromwell/jobs/__main__.py

```diff
@@ -31,13 +33,42 @@
     args, _ = parser.parse_known_args()
 
 app = connexion.App(__name__, specification_dir='./swagger/', swagger_ui=False)
+DEFAULT_CROMWELL_CREDENTIALS = {"cromwell_user": "", "cromwell_password": ""}
+LOADER_ERR_MSG = "Failed to load config.json, using the default config: {0}".format(
+    DEFAULT_CROMWELL_CREDENTIALS)
+
+# Load credentials for cromwell
 config_path = os.environ.get('CROMWELL_CREDENTIALS')
-with open(config_path) as f:
-    config = json.load(f)
+
+# Check if the credentials are provided properly
+try:
+    with open(config_path) as f:
+        config = json.load(f)
+except (IOError, TypeError):
+    print(LOADER_ERR_MSG)
+    config = DEFAULT_CROMWELL_CREDENTIALS
+finally:
     app.app.config.update(config)
+
 app.app.config['cromwell_url'] = args.cromwell_url
 app.app.json_encoder = JSONEncoder
 app.add_api('swagger.yaml', base_path=args.path_prefix)
 
-if __name__ == '__main__':
-    app.run(host='0.0.0.0', port=args.port)
+
+def run():
```

Comment:
Does this need to be in a separate method? If you put these checks on the top-level and left the entrypoint as `jobs.__main__:app` does that not work? Fine if not just wanted to check.

Reply:
The goal here is to provide a startup check, if it fails, the Cromwell shim shouldn't even start. 

It might because I haven't found a good way for this, but if we point the gunicorn to`jobs.__main__:app`, even the checks failed, gunicorn can still find the application(global var `app.app`) and start it. But the first request to the shim API will then invoke the checks and the shim layer will blow up with thousands of same error messages :/

Flask's Resolvers may deal with this case but it looks more complicated at the first glance. So I moved the `app` var to `__init__.py` here to provide a feasible startup check. Please comment on here if there is a better way to deal with this.

Reply:
Gotcha, didn't realize the app would still run if an exception was thrown in `__main__`. This sounds like a reasonable solution to me, thanks for the explanation!